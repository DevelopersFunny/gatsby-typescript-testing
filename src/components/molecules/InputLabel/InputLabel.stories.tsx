import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'
import { InputLabel } from '.'

export default {
  title: 'Atoms/InputLabel',
  component: InputLabel,
} as ComponentMeta<typeof InputLabel>

const Template: ComponentStory<typeof InputLabel> = args => (
  <InputLabel {...args} />
)

export const InputLabelName = Template.bind({})
InputLabelName.args = {
  name: 'inputlabel',
  id: 'inputlabel',
  type: 'text',
  placeholder: 'Input label',
}
