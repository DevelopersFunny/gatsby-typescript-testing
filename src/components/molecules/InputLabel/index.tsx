import React from 'react'
import { IInputLabelProps } from '../../../types/components/input'
import { Input } from '../../atoms/input'
import { Label } from '../../atoms/Label'

export const InputLabel = ({
  id,
  maxLength,
  minLength,
  name,
  type,
  label,
  disabled,
  onChange,
}: IInputLabelProps) => {
  return (
    <>
      <div className="relative">
        <Input
          id={id}
          name={name}
          type={type}
          placeholder={label}
          minLength={minLength}
          maxLength={maxLength}
          className="peer"
          disabled={disabled}
          onChange={onChange}
        />
        <Label
          htmlFor={id}
          className="
            absolute
            -top-3.5
            h-10
            w-full pl-9
            pr-3
            text-sm
            text-gray-600 
            transition-all 
            peer-placeholder-shown:top-2
            peer-placeholder-shown:text-base 
            peer-placeholder-shown:italic 
            peer-placeholder-shown:text-gray-400
            peer-focus:-top-3.5 
            peer-focus:text-sm 
            peer-focus:text-gray-600"
        >
          {label}
        </Label>
      </div>
    </>
  )
}
