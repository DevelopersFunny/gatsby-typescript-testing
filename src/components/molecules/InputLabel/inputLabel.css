.inputStyled {
  @apply h-10 w-full border-b-2 border-gray-300 text-gray-900 focus:outline-none focus:border-red-600;
}

.labelStyled {
  @apply peer-placeholder-shown:uppercase;
}
