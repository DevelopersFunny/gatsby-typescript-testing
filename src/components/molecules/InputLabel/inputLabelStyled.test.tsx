import React from 'react'
import * as stories from './InputLabel.stories'
import { composeStories } from '@storybook/testing-react'
import { render, screen } from '@testing-library/react'
import { createSerializer, matchers } from '@emotion/jest'

/**
 * Emotion matchers and snapshots
 */
expect.extend(matchers)
expect.addSnapshotSerializer(createSerializer())

/**
 * Stories render
 */
const { InputLabelName } = composeStories(stories)

describe('Input render with props', () => {
  it('should render input', () => {
    render(
      <InputLabelName name="test" id="test" label="Input Label" type="text" />
    )
    const inputLabelElement = screen.getByText('Input Label')
    expect(inputLabelElement).toMatchSnapshot()
  })
})
