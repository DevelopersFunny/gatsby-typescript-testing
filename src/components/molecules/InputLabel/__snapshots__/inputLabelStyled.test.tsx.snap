// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input render with props should render input 1`] = `
.emotion-0 {
  display: block;
  font-size: 0.875rem;
  line-height: 1.25rem;
  font-weight: 500;
  --tw-text-opacity: 1;
  color: rgba(55, 65, 81, var(--tw-text-opacity));
}

.emotion-0 {
  display: block;
  font-size: 0.875rem;
  line-height: 1.25rem;
  font-weight: 500;
  --tw-text-opacity: 1;
  color: rgba(55, 65, 81, var(--tw-text-opacity));
}

<label
  class=" absolute -top-3.5 h-10 w-full pl-9 pr-3 text-sm text-gray-600  transition-all  peer-placeholder-shown:top-2 peer-placeholder-shown:text-base  peer-placeholder-shown:italic  peer-placeholder-shown:text-gray-400 peer-focus:-top-3.5  peer-focus:text-sm  peer-focus:text-gray-600 emotion-0"
  for="test"
>
  Input Label
</label>
`;
